index.rst

```diff
@@ -13,17 +13,188 @@
+Note that currently mobu caches notebooks on start-up and needs to be restarted to pick up new notebooks; an API call or other method to induce a running mobu to re-poll its notebook repos would be useful (but not critical).
+
+Reliance on specific data holdings
+----------------------------------
+
+**Before:** System-test notebooks address specific data holdings
+
+**After:** Notebooks perform a data discovery step and run on arbitrary holdings and/or opt out of data-holding specific checks.
+
+**Discussion:**
+
+From the beginning we have identified the need to have a small data-set that is available on all deployments to allow system-test notebooks to run everywhere.
+While there is merit to this idea, in practice finding the effort to curate such a careful minimal in size but maximal in utility dataset has been hard to find.
+With the advent of the tutorial-notebooks repo, the requirement for performing substantive computations and/or service load has been eliminated from system-test.
+With the proposed data curation notebooks, that require specific data holdings can live elsewhere.
+To the extent that this is practical, system-test service notebooks should be written with a data discovery or data check step to see what data is available (eg. in terms of available catalogs, tap_schema could be queried first to make sure unavailable catalogs are not being requested).
+In practice this is may be an aspirational requirement.
+
+Branches
+--------
+
+**Before:** Mobu typically runs the prod branch of the notebooks (though this is configurable) with notebooks having to be cherry-picked from main to prod.
+
+**After:** End this madness.
+
+**Discussion:**
+
+The need to maintain two different branches has been eliminated with mobu's ability to easily be configured to run off different branches for cases where it is useful to have an "in-development" version deployed.
+Hence cherry-picking is just annoying with no particular benefit.
+
+Outputs
+-------
+
+**Before:** Notebook contributors need to remember to clear outputs before checking in new versions of the notebooks.
+
+**After:** Have this happen automatically (via pre-commit hook or similar), or at least raise a CI warning if there is output checked in.
+
+**Discussion:**
+
+Having the human remember to clear outputs before saving and checking in is error prone. Even if the notebook ends in clear_outputs(), it still implies it was run to the end before commit.
+Ideally something like https://github.com/srstevenson/nb-clean would be integrated in the development workflow.
+
+This may also be of use to other notebook repo maintainers.
+
+
+Write-Only
+----------
+
+**Before:** Notebooks are checked out write-only in nublado to avoid conflicts
+
+**After:** Tutorial-notebooks continue to be write-only (this has saved a lot of support headaches) but system-test notebooks can be read-write to allow for easier guided troubleshooting.
```

Comment:
How do we want to handle user modifications to system-test notebooks when we try to check out the latest version during notebook startup?

Reply:
Yeah good question.  Options:

1. check out system-test in a dot directory (so at least that it's not going to be found naively)
2. always wipe out system-test and user changes are lost on spin-up (clearly indicated)
3. add a JL option to force-refresh system-test and if anyone ends up in a support call tell them to do that first
4. I am wrong about this and the present way is fine

Thoughts?

Reply:
What about not checking out system-test by default and instead providing a menu item that checks it out with a forced update to the latest version? I guess the drawback of that is that they don't get system-test if their lab is broken enough that the menu item doesn't work, but in that case we could still tell them how to check it out manually, so I'm not sure much is lost.

If you want everyone to always have it, I think always wiping it out would be fine. We're the only people likely to modify those notebooks and we know to do development elsewhere.

Reply:
Okay yes, we can add a menu item that populates system-test (also less likely. to confuse users who find them on their filesystem). (Mobu test users check them directly from Github so won't be affected by this). [the menu option would still have to check and overwrite any that were already checked out]

Reply:
https://github.com/lsst-sqre/sqr-080/commit/a59c423247473244192dafc17ea00df115c1909a